### PR TITLE
React on rails setup

### PR DESCRIPTION
**Pull request by GergKllai1**
# PT Link

[React on rails setup](https://www.pivotaltracker.com/story/show/163532907)

# Description of PR

- Sets up React on Rails gem
- Sets up the project structure
- Cleans up code

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -10,6 +10,8 @@ gem 'uglifier', '>= 1.3.0'
 gem 'webpacker'
 gem 'turbolinks', '~> 5'
 gem 'bootsnap', '>= 1.1.0', require: false
+gem 'react_on_rails', '11.1.4'
+
```
Comment (HannaSwDn) [inline Gemfile:14]: remove this line please

File: app/views/layouts/main.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
```
Comment (HannaSwDn) [inline app/views/layouts/main.html.erb:1]: Are we going to use HAML or erb?

Comment (GergKllai1) [reply app/views/layouts/main.html.erb:1]: I would stick to ERB, we will not use that much of it anyhow

**Review by kianaditya — APPROVED** [review]
LGTM!

**Review by schneiderman805 — APPROVED** [review]
Hanna did a good review, I cant seem to see anything that needs changing.

File: app/views/main/index.html.erb [hunk 3]
```diff
@@ -0,0 +1 @@
+<%= react_component("Main", props: @main_props, prerender: true) %>
```
Comment (magnus-thor) [inline app/views/main/index.html.erb]: `<%= react_component("Main", props: @main_props, prerender: Rails.env.test?) %>`

**Review by HannaSwDn — APPROVED** [review]

**Pull request merged**